go/enclave/evm/evm_facade.go

```diff
@@ -38,7 +39,7 @@ func ExecuteTransactions(txs []*common.L2Tx, s *state.StateDB, header *common.Ba
 	}
 
 	for i, t := range txs {
-		r, err := executeTransaction(s, chainConfig, chain, gp, ethHeader, t, usedGas, vmCfg, fromTxIndex+i)
+		r, err := executeTransaction(s, chainConfig, chain, gp, ethHeader, t, usedGas, vmCfg, fromTxIndex+i, header.Hash())
```

Comment:
could enrich the receipt with the hash at this layer rather than pass the hash in just so it can be added to the receipt

Reply:
it's an option, but it felt more straight forward to do it when the receipt is first returned